crates/uv-build-backend/src/lib.rs

```diff
@@ -336,6 +344,51 @@ pub fn metadata(source_tree: &Path, metadata_directory: &Path) -> Result<String,
     Ok(dist_info_dir)
 }
 
+/// PEP 517 requires that the metadata directory from the prepare metadata call is identical to the
+/// build wheel call. This method performs a prudence check that `METADATA` and `entry_points.txt`
+/// match.
+fn check_metadata_directory(
+    source_tree: &Path,
+    metadata_directory: Option<&Path>,
+    pyproject_toml: &PyProjectToml,
+) -> Result<(), Error> {
+    let Some(metadata_directory) = metadata_directory else {
+        return Ok(());
+    };
+
+    let dist_info_dir = format!(
+        "{}-{}.dist-info",
+        pyproject_toml.name().as_dist_info_name(),
+        pyproject_toml.version()
+    );
+
+    let metadata = pyproject_toml
+        .to_metadata(source_tree)?
+        .core_metadata_format();
+    if fs_err::read_to_string(metadata_directory.join(&dist_info_dir).join("METADATA"))? != metadata
+    {
+        return Err(Error::InconsistentSteps("METADATA"));
+    }
+
+    let entrypoints_path = metadata_directory
+        .join(&dist_info_dir)
+        .join("entry_points.txt");
+    match pyproject_toml.to_entry_points()? {
+        None => {
+            if entrypoints_path.is_file() {
+                return Err(Error::InconsistentSteps("entry_points.txt"));
+            }
+        }
+        Some(entrypoints) => {
+            if fs_err::read_to_string(&entrypoints_path)? != entrypoints {
+                return Err(Error::InconsistentSteps("entry_points.txt"));
+            }
+        }
+    }
```

Comment:
How come there's asymmetry  between the `entry_points.txt` checks and `METADATA`? For `METADATA`, you check for content equivalence, and error if the file doesn't exist. But here, an error is returned if the file does exist.

OK, maybe writing that out helped me understand. It looks like creating entry points from `pyproject_toml` might return `None`, but that case doesn't exist for `to_metadata`?

Apologies for the meandering comment. :)

Reply:
I've added inline comments: METADATA is mandatory and always written, `entry_points.txt` is not written if it would be empty.

Reply:
Thanks!